src/node_options.cc

```diff
@@ -113,19 +114,61 @@ void EnvironmentOptions::CheckOptions(std::vector<std::string>* errors) {
 
 namespace options_parser {
 
-// Explicitly access the singelton instances in their dependancy order.
-// This was moved here to workaround a compiler bug.
-// Refs: https://github.com/nodejs/node/issues/25593
+class DebugOptionsParser : public OptionsParser<DebugOptions> {
+ public:
+  DebugOptionsParser();
+};
+
+class EnvironmentOptionsParser : public OptionsParser<EnvironmentOptions> {
+ public:
+  EnvironmentOptionsParser();
+  explicit EnvironmentOptionsParser(const DebugOptionsParser& dop)
+    : EnvironmentOptionsParser() {
+    Insert(dop, &EnvironmentOptions::get_debug_options);
+  }
+};
 
-#if HAVE_INSPECTOR
-const DebugOptionsParser DebugOptionsParser::instance;
-#endif  // HAVE_INSPECTOR
+class PerIsolateOptionsParser : public OptionsParser<PerIsolateOptions> {
+ public:
+  PerIsolateOptionsParser() = delete;
+  explicit PerIsolateOptionsParser(const EnvironmentOptionsParser& eop);
+};
 
-const EnvironmentOptionsParser EnvironmentOptionsParser::instance;
+class PerProcessOptionsParser : public OptionsParser<PerProcessOptions> {
+ public:
+  PerProcessOptionsParser() = delete;
+  explicit PerProcessOptionsParser(const PerIsolateOptionsParser& iop);
+};
 
-const PerIsolateOptionsParser PerIsolateOptionsParser::instance;
+#if HAVE_INSPECTOR
+const DebugOptionsParser DebugOptionsParserInstance{};
```

Comment:
Can you put these into `namespace per_process` (in the top scope)?

Reply:
It's not simple since they are instances of type declared only in this file.
I renamed them to emphasize they are private to this translation unit.